src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/DeprecationControl.xaml

```diff
@@ -0,0 +1,83 @@
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+             mc:Ignorable="d"
+             d:DesignHeight="150" d:DesignWidth="350"
+             Background="{DynamicResource {x:Static nuget:Brushes.InfoBackgroundKey}}">
+  <UserControl.Resources>
+    <ResourceDictionary>
+      <ResourceDictionary.MergedDictionaries>
+        <nuget:SharedResources />
+      </ResourceDictionary.MergedDictionaries>
+    </ResourceDictionary>
+  </UserControl.Resources>
+  <UserControl.CommandBindings>
+    <CommandBinding
+      Command="{x:Static nuget:PackageManagerControlCommands.OpenExternalLink}"
+      Executed="ExecuteOpenExternalLink" />
+  </UserControl.CommandBindings>
+  <Grid>
+    <Expander
+      Margin="0,12,0,16"
+      IsExpanded="True"
+      Foreground="{DynamicResource {x:Static nuget:Brushes.UIText}}">
+      <Expander.Header>
+        <TextBlock
+        FontWeight="Bold">
+          <Run
+            FontWeight="Bold"
+            Text="{x:Static nuget:Resources.Label_Deprecated}" />
+          <Run Text="{Binding Path=PackageDeprecationReasons}" />
+        </TextBlock>
+      </Expander.Header>
+      <StackPanel
+        Orientation="Vertical">
+        <TextBlock
+          Margin="0,8,0,0"
+          TextWrapping="WrapWithOverflow"
+          TextTrimming="WordEllipsis"
+          Text="{Binding Path=PackageDeprecationReasonsExplained}" />
+        <StackPanel
+          Margin="0,8,0,0"
+          Orientation="Vertical"
+          Visibility="{Binding Path=PackageDeprecationAlternatePackageLinkText, Converter={StaticResource NullToVisibilityConverter}}">
+          <TextBlock
+            Text="Alternate package"
+            FontWeight="Bold"/>
+          <StackPanel
+            Orientation="Horizontal"
+            VerticalAlignment="Center">
+            <TextBlock
+              TextWrapping="WrapWithOverflow"
+              VerticalAlignment="Center">
+              <Hyperlink
+                NavigateUri="{Binding Path=PackageDeprecationAlternatePackageLinkUrl}"
+                Style="{StaticResource HyperlinkStyle}"
+                Command="{x:Static nuget:PackageManagerControlCommands.OpenExternalLink}">
+                <Run Text="{Binding Path=PackageDeprecationAlternatePackageLinkText}" />
+              </Hyperlink>
+            </TextBlock>
+            <TextBlock
+              Text="{x:Static nuget:Resources.Label_OrAbove}"
+              VerticalAlignment="Center"
+              Visibility="{Binding Path=PackageDeprecationAlternatePackageRangeHasUpperBound, Converter={StaticResource BooleanToVisibilityConverter}}"/>
+          </StackPanel>
+        </StackPanel>
+        <StackPanel
+          Margin="0,8,0,0"
+          Visibility="{Binding Path=PackageDeprecationCustomMessage, Converter={StaticResource NullToVisibilityConverter}}">
+          <TextBlock
+            Text="Package authors message"
```

Comment:
author's

Reply:
If you consider another feed implementing deprecation, we actually have no idea who deprecated the package. It could be an automated process from the feed maintainers. It could be IT at a big corporation. Maybe it should just say "Message".

Reply:
The gallery UI says "Additional details"...I think we should align on that here.

Reply:
Changed to "Additional details" so it is consistent with the gallery UI. https://github.com/NuGet/NuGet.Client/pull/2899/commits/c807e4fc75ffb11f6541ac652dca2d6701dcbc94